apis/v1alpha1/gateway_types.go

```diff
@@ -234,6 +234,41 @@ type Listener struct {
 	//
 	// Support: Core
 	Routes RouteBindingSelector `json:"routes"`
+
+	// BackendRef is a local object reference to route requests to.
+	//
+	// If the referent cannot be found, the controller should raise
+	// the "ResolvedRefs" condition on the Gateway with the "DegradedListeners"
+	// reason. The gateway status should be updated with a condition
+	// that describes the error more specifically.
+	//
+	// Support: Custom
+	//
+	// +optional
+	BackendRef *GatewayBackendRef `json:"backendRef,omitempty"`
```

Comment:
This seems inconsistent with route's forward to, which is
```
forwardTo:
  service: ..
  backendRef: ...
  port: ...
```

Since now we have a backendRef with service and port inside of it?

We also have backendRef in backendpolicy which is different as well, which I worry may confuse users

Reply:
I agree with you @howardjohn. I'll add that back since it felt weird to not have it there, however, it also seemed like this was a simple shortcut path proposal which might warrant not having it there.

Reply:
I think not including `backendRef` is better. Let's start with the bare minimum and then include other resources as required.
Custom backends at this level does make me think if we are adding one more extension point in the Gateway resource than needed. This is really a shortcut for a specific problem and I say let's keep it short.

Reply:
I agree that we should target this as narrowly as possible. Seeing it here, I think `backendRef` could get confusing since it's so much narrower than backendRef elsewhere. I think we have a few options here:

1. Introduce `listener.forwardTo` or `listener.backend` that would include `serviceName`, `backendRef`, and `port` fields. This is the most consistent with the rest of the API and would enable a great deal of functionality, but at the cost of increased complexity and too many ways to do something. IE how long until we'd want to bubble up filters or other route concepts to listeners?
2. Update `listenener.routes.kind` to support `Service`. This gets rather confusing as it's no longer a direct reference and Services are definitely not Routes.
3. Tweak this approach to define `listener.service` with `name` and `port` fields. 

I think 3 is definitely the simplest and most straightforward option here. One concern I have is that it would exclude referencing multicluster Services at this level. Since this is intended to be a shortcut, that may not be a significant issue, but want to be sure. 

I've also seen some discussion over the past year of updating route selection to support a Service type. I don't think we'd want to do both, so we really need to be sure we don't want listeners to be able to select Services in the same way they can select backends.

/cc @JeremyOT @mark-church @akutz

Reply:
There's two different dimensions that we're discussing here:

- The ability for a Gateway to target multiple L4 resources vs a single L4 resource
- The ability for a Gateway to target just Services vs other CRs as well

There are valid use-cases across all of these, though I definitely do not see the use-cases all as the same priority:

1. Highest-value use-case and the one that started this design decision is the ability to map a single Service directly to a Gateway. I think that for Gateway to have any success at all for L4 this must be done, because otherwise the verbosity of creating an empty UDP/TCPRoute makes it feel too complex to the average user.
2. Another use-case mentioned was to map a single multi-cluster Service (think of a hypothetical CR that represents an aggregation of Services across clusters) to a Gateway.
3. There's another interesting case that is more relevant for service mesh use-cases which is L7 Service virtualization. This is where the Service FQDN is taken to generate an implicit host route for that Service. This allows a Service to automatically be routed at L7 without any additional configuration. This requires the mapping of multiple Services (using `listener.routes.kind=Service`) to a single Gateway where they are differentiated via their "virtualized" Service hostnames.

In all three cases we are actually doing the same thing: creating a Route shortcut. 

For #1 it creates an empty TCPRoute shortcut to a Service. For #2 it creates an empty TCPRoute shortcut to a CR. For #3 it creates an empty HTTPRoute shortcut (using the Service FQDN) to a Service. I originally thought of these as separate things but after thinking about it, this is all closely related and logically consistent.

I agree that we should not open up the possibilities too soon because if we move too fast without considering the consequences then we could start to create some inconsistencies. We should continue to prioritize use-case #1 but it would be great if it would not block #2 or #3 in the future. 

It feels like a `listener.forwardTo` which only includes `serviceName` and `port`, but not `backendRef` would be consistent but not open up too much. `listener.routes.kind=Service` could always be added in the future to support Service virtualization.

Reply:
It seems like we have 2 primary concerns here:

1. A meaningful name that doesn't get confused with other similar fields in the existing API
2. Ensuring this does not take on the full scope of `forwardTo` and instead is reserved for use as a simple shortcut

So with that background, I don't think we should reuse either `backendRef` or `forwardTo` terms as they both have different meanings and larger scopes than what we're proposing here. Since this is essentially a route shortcut, what if we used `routeTo`? That would include `serviceName` and `port`, and it would be fairly straightforward to explain that the `routes` and `routeTo` fields were mutually exclusive fields.